apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -888,36 +905,119 @@ type FailedResourcePlacement struct {
 	Condition metav1.Condition `json:"condition"`
 }
 
-// DriftedResourcePlacement contains the drifted resource details.
+// JSONPatchOp is the operation of a JSON patch. Fleet uses JSON patches to describe drifts
+// and configuration differences found in the resources; in the case of Fleet, only
+// the Add, Remove, and Replace operations are used.
+// +enum
+type JSONPatchOp string
+
+const (
+	// JSONPatchOpAdd is the Add operation in a JSON patch.
+	JSONPatchOpAdd JSONPatchOp = "add"
+
+	// JSONPatchOpRemove is the Remove operation in a JSON patch.
+	JSONPatchOpRemove JSONPatchOp = "remove"
+
+	// JSONPatchOpReplace is the Replace operation in a JSON patch.
+	JSONPatchOpReplace JSONPatchOp = "replace"
+)
+
+// JSONPatchDetail describes a JSON patch that explains an observed configuration drift or
+// difference.
+type JSONPatchDetail struct {
+	// Op is the JSON patch operation.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Enum=add;remove;replace
+	Op JSONPatchOp `json:"op"`
+
+	// The JSON path that points to a field that has drifted or has configuration differences.
+	// +kubebuilder:validation:Required
+	Path string `json:"path"`
+
+	// The current value at the JSON path, i.e., the value before the JSON patch operation.
+	//
+	// For the JSON patch operation Add, this field will be left empty.
+	//
+	// Note that this field is kept for informational purposes only; it is not a required part of the
+	// JSON patch; see RFC 6902 for more information.
+	// +kubebuilder:validation:Optional
+	CurrentValue string `json:"value,omitempty"`
```

Comment:
I am wondering whether you still need the "op" if you show both values. BTW, CurrentValue vs Value is confusing. Can we use hub value and member value?

Path: "/name" ValueInHub: "foo" ValueInMember: "bar"
Path: "/age" ValueInHub: "10" ValueInMember: nil
Path: "/occupation" ValueInHub: nil ValueInMember: "engineer"

Reply:
Thanks Liqian! I have made the requested changes. 🙏